core/controllers/classroom_test.py

```diff
@@ -231,6 +231,144 @@ def test_get(self) -> None:
+        topic_services.save_new_topic(admin_id, no_summary_topic)
+        topic_services.delete_topic_summary(no_summary_topic_id)
+
+        self.login(self.CURRICULUM_ADMIN_EMAIL, is_super_admin=True)
+        self.save_new_valid_classroom(
+            classroom_id='test_id',
+            topic_id_to_prerequisite_topic_ids={
+                        no_summary_topic_id: [],
+                        self.public_topic_id_1: [],
+                        self.private_topic_id: []
+            },
+            course_details='Course details for classroom.',
+            topic_list_intro='Topics covered for classroom'
+        )
+        self.logout()
+
+        json_response = self.get_json(
+            '%s/%s' % (feconf.CLASSROOM_DATA_HANDLER, 'math'))
+        topic_summary_dict = (
+            topic_fetchers.get_topic_summary_by_id(
+                self.public_topic_id_1
+            ).to_dict()
+        )
+        public_topic_1_summary_dict_is_published = {'is_published': True}
+        public_topic_1_summary_dict = dict(
+            topic_summary_dict,
+            **public_topic_1_summary_dict_is_published
+        )
+
+        topic_summary_dict = (
+            topic_fetchers.get_topic_summary_by_id(
+                self.private_topic_id
+            ).to_dict()
+        )
+        private_topic_summary_dict_is_published = {'is_published': False}
+        private_topic_summary_dict = dict(
+            topic_summary_dict,
+            **private_topic_summary_dict_is_published
+        )
+
+        # Skips 'no_summary_topic'.
+        expected_dict = {
+            'classroom_id': 'test_id',
+            'name': 'math',
+            'url_fragment': 'math',
+            'topic_summary_dicts': [
+                public_topic_1_summary_dict, private_topic_summary_dict
+            ],
+            'course_details': 'Course details for classroom.',
+            'topic_list_intro': 'Topics covered for classroom',
+            'teaser_text': 'Teaser Text',
+            'thumbnail_data': dummy_thumbnail_data.to_dict(),
+            'banner_data': dummy_banner_data.to_dict(),
+            'is_published': True,
+            'public_classrooms_count': 1
+        }
+        self.assertDictContainsSubset(expected_dict, json_response)
+
+    def test_get_multiple_classrooms_count_all_published_unmatching_classrooms(
+        self
+    ) -> None:
+        self.login(self.CURRICULUM_ADMIN_EMAIL, is_super_admin=True)
+        # Create a published classroom with matching url_fragment with 'math'.
+        self.save_new_valid_classroom(
+            classroom_id='test_id',
+            topic_id_to_prerequisite_topic_ids={
+                        self.public_topic_id_1: [],
+                        self.private_topic_id: []
+            },
+            course_details='Course details for classroom.',
+            topic_list_intro='Topics covered for classroom'
+        )
+        # Create a unpublished classroom with unmatching url_fragment 'math'.
```

Comment:
... with a URL fragment that does not match 'math'.

Reply:
Thanks for the suggestion. It's been updated